pkg/reconciler/service/workerpool.go

```diff
@@ -16,10 +23,12 @@ type workPoolBuilder struct {
 }
 
 type WorkerPool struct {
-	debug        bool
-	logger       *zap.SugaredLogger
-	antsPool     *ants.Pool
-	newRunnerFct func(context.Context, *reconciler.Task, callback.Handler, *zap.SugaredLogger) func() error
+	debug                bool
+	logger               *zap.SugaredLogger
+	antsPool             *ants.Pool
+	PoolID               string
+	occupancyCallbackURL string
```

Comment:
We should think of using a push instead of a pull model: the occupancy-collector module in the mothership reconciler is requesting the occupancy per reconciler.

This scales much better instead letting each reconciler report it to the mothership.